api/v1alpha1/envoyfleet_types.go

```diff
@@ -166,6 +166,16 @@ type TLS struct {
 	// You can specify more than one
 	// For more information on how certificate selection works see: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl#certificate-selection
 	TlsSecrets []TLSSecrets `json:"tlsSecrets"`
+
+	// +optional
+	// TLS requirement for hosts managed by EnvoyFleet
+	// NONE - No TLS requirement for the virtual hosts
+	// EXTERNAL_ONLY - External requests must use TLS. If a request is external and it is not
+	//	using TLS, a 301 redirect will be sent telling the client to use HTTPS.
+	// ALL - All requests must use TLS. If a request is not using TLS, a 301 redirect
+	//	will be sent telling the client to use HTTPS.
+	// Defaults to NONE
+	Requirement string `json:"requirement"`
```

Comment:
I think this setting should be on/off - given that we're the gateway and user just need to give "yes/no" decision if the traffic should be automatically redirected. This also should be user friendly, the "Requirement" is a vague keyword. I propose json tag `https_redirect:`  which will be the list of vhosts it applies to. We cannot enable automatic https redirect for all enabled vhosts, since this means one setting affects all API/Static documents (sites) in the cluster (given it has only one default fleet). User should at least specify the list of vhosts this setting applies to.

Reply:
in that case then, this is what I proposed before, our host type which is just an alias to a string, we could turn that into a struct with the host name as a string field and a https_redirect bool field

Reply:
I agree, we can do that in the inner type, but not expose to the users in API or StaticRoute objects.
To clarify, we have a weak place in our UX, when users can set vhost (domain name) in each API or StaticRoute documents and this can cause conflicts if there are multiple teams deploying multiple objects with matching vhost names or those that have wildcards.
The same will apply to this automatic https settings as well, i.e. one team would not want the automatic redirect for the domain in the object vhosts list ([example1.com, example2.com]) while other team wants that for e.g. example2.com. That's why I think we should be specific for this setting and set it in EnvoyFleet, where there couldn't be duplicates.
During the configuration parsing we then can see if the vhost domain matches that EnvoyFleet list and enable the redirect in the inner type.